rest-client/src/test/java/org/sagebionetworks/bridge/rest/ErrorResponseInterceptorTest.java

```diff
@@ -122,24 +125,25 @@ public void badRequest400() throws Exception {
     @Test
     public void unrecognizedUrl400() throws Exception {
         // Play sends responses like these if you call an unrecognized URL.
-        doReturn(400).when(response).code();
-        doReturn("<html><body>Play returns HTML instead of JSON</body></html>").when(body).string();
-        doReturn("Play Framework does not recognize this URL").when(response).message();
+        doReturn(404).when(response).code();
+        String json = Tests.unescapeJson("{'message':'Endpoint not found.','type':'EndpointNotFoundException'}");
+        doReturn(json).when(body).string();
```

Comment:
Similarly, even though we've fixed one specific case where the server returns HTML, do we want the Java SDK to handle this case in general, for the sake of robustness? Off the top of my head, I know some 503 Server Unavailable are likely to return HTML or some other non-JSON doc, and because of how load balancers work, this is in some cases unavoidable.